### DISCUSSION TITLE
build errors 0.8.1: some optional dependencies not optional?

### DISCUSSION THREAD

**Discussion by jonleivent** [labwc/labwc#2423] [Q&A] [answered]
From https://github.com/labwc/labwc/issues/2269#issuecomment-2539854563

Sorry about the misplaced post...

I can't build 0.8.1. Attached are my config and the build errors. I'm on Debian trixie.

[setup.txt](https://github.com/user-attachments/files/18116607/setup.txt)
[build.txt](https://github.com/user-attachments/files/18116608/build.txt)

Perhaps some of the optional dependencies I omitted are not optional?

**Comment by Consolatis** [comment]
Hm.. labwc should indeed check for some wlroots defines. The reason it fails is that wlroots basically got no backends:
```
wlroots 0.18.1

    drm-backend     : NO
    x11-backend     : NO
    libinput-backend: NO
    xwayland        : NO
    gles2-renderer  : NO
    vulkan-renderer : NO
    gbm-allocator   : NO
    session         : NO
    color-management: NO
    xcb-errors      : NO
    egl             : NO
    libliftoff      : NO
```

```
wlroots| Build-time dependency hwdata found: NO (tried pkgconfig and cmake)
wlroots| Message: Required for the DRM backend.
```

```
wlroots| Run-time dependency gbm found: NO (tried pkgconfig and cmake)
```

```
wlroots| Run-time dependency libseat found: NO (tried pkgconfig and cmake)
wlroots| Looking for a fallback subproject for the dependency libseat
wlroots| Neither a subproject directory nor a seatd.wrap file was found.
wlroots| Subproject  seatd is buildable: NO (disabling)
wlroots| Dependency libseat from subproject seatd found: NO (subproject failed to configure)
wlroots| Message: Required for session support.
```

I am not sure why the `libinput-backend` and `gles2-renderer` wasn't found though.

From my `dpkg-query --list`:
- hwdata
- libseat-dev
- libgbm-dev

**Reply by jonleivent** [reply]
I note that labwc 0.7.4-1 is in Debian trixie's repo.  Which means someone has the full recipe (all other required apt packages) for how to build it in trixie.  Is 0.8.1 buildable with the same recipe - and if so, who has that recipe?  For each version of something 0.8.1 requires that is not in trixie, is there an appropriate subproject wrap file?

I'd prefer to build a minimal labwc 0.8.1, but I'll take any.

**Reply by Consolatis** [reply, marked as answer]
As you run Trixie (which I assume is the current Testing) this should work: https://github.com/labwc/labwc/blob/master/.github/workflows/build.yml#L76

**Reply by jonleivent** [reply]
Thanks!  That worked (specifically, the two `apt-get build-dep`'s).

**Comment by Consolatis** [comment]
Follow-up:
- #2433

**Comment by jonleivent** [comment]
Is it possible to have the configure step determine the required (vs optional) dependencies based on some user configuration setting?

Because I didn't want a headless or nested-only build.  But I also didn't want to install every optional dependency.

**Reply by jonleivent** [reply]
Oh, never mind.

This just requires some documentation about what functionality each optional dependency adds.
